### PR TITLE
Move histogram functions out of toolkit.lisp.

### PR DESCRIPTION
**Pull request by rpgoldman**
Most of toolkit.lisp must be loaded early (macros, etc.), but the histogram functions use random, so the dependencies are much simpler if we pull them out (into new histogram.lisp) and push them down in the load order.

### CODE REVIEW

**Comment by rpgoldman** [comment]
Probably best to merge the build clean test first. Then I will rebate this and we can be certain it achieves its purpose before merging.

File: toolkit.lisp [hunk 1]
```diff
@@ -66,93 +66,4 @@
...
-(defmacro %arr (type &rest elements)
-  `(make-array ',(list-dim elements)
-               :element-type ',type
-               :initial-contents ',elements))
-
-(defmacro define-pregenerated (name contents)
-  `(progn
-     (let ((contents ,contents))
-       (defun ,name () contents))
-     (define-symbol-macro ,name (,name))))
```
Comment (Shinmera) [inline toolkit.lisp:158]: why are all these functions removed?

Comment (rpgoldman) [reply toolkit.lisp:158]: That is a very good question. My best guess is that I did it inadvertently when I rebased this change on top of your latest master. I will go back and fix it.

Comment (rpgoldman) [reply toolkit.lisp]: I think this is fixed now.

**Comment by rpgoldman** [comment]
OK, I see how that happened. I had *another* commit in a different location that moved the `copy` methods out of toolkit. The reason I did that was that the `copy` methods were getting loaded before the `defgeneric` for `copy`.

We might want to move these 2 defmethods for `copy` to below the `def-generator-fun` for `copy` in generator.lisp

**Comment by rpgoldman** [comment]
OK, I can make a separate PR for moving the `copy` method definitions.

**Pull request merged**